### DISCUSSION TITLE
can response body limit size？

### DISCUSSION THREAD

**Discussion by lrfdehao** [apache/apisix#5892] [Q&A]
Suppose I have two fields, `include_resp_body` as the switch, `resp_limit_size` as the limit size. After configuring these two parameters, if resp_body exceeds the size of resp_limit_size, resp_body will not be recorded in the log. The same is true for req_body

**Comment by lrfdehao** [comment]
在日志打印的时候，能否新增一个 resp_limit_size 的字段来控制响应 body 的打印？例如响应 body 超过了这个长度，则默认不打印，或者按规定长度截断这个 resp_body？

**Reply by tzssangglass** [reply]
you can use `{"request_length", "<", 1024}` in include_req_body_expr and `{"http_content_length", "<", 1024}` in include_resp_body_expr to limit size, but  you need know that `request_length` also includes the headers.

**Reply by tzssangglass** [reply]
see: https://github.com/apache/apisix/pull/5886/files#diff-3c54d9c32b29d45c590bb52a1f5986de6e4caf4001277a242bdf511e5d66a6feR630-R638

**Reply by lrfdehao** [reply]
我明白你的意思，但是我这边想要的是全局一个可控的变量，不单单指的是 kafka-logger 这个插件，因为我理解不是所有的请求的响应都有 content-length 这个参数的

**Reply by lrfdehao** [reply]
比方说一个请求的响应没有 content-length 来告知 resp body 的长度，那是否就需要在获取 resp_body 的地方去对 resp_body 做长度计数然后判断，这样是不是就没有办法通过表达式的方式去控制这个 resp body 的打印了

**Reply by tzssangglass** [reply]
LGTM, although we can also do this with custom Nginx variables, it is easier to do it in a plugin.

**Reply by tzssangglass** [reply]
the request body also requires this

**Reply by lrfdehao** [reply]
其实我针对这一次场景做了一次提交，是借助了 nginx 多次读取 resp_body 的特性，在 core.response 中对 resp_body 做了处理与判断：https://github.com/lrfdehao/apisix/commit/0672ca0a35e5f72f70f2a2edb2cfd54f0c2e0764

**Reply by lrfdehao** [reply]
我是第一次搞这些，也不太清楚流程和规范，如果方便的话，可以加个微信交流一下吗

**Reply by tzssangglass** [reply]
hi @juzhiyuan PLAT

**Reply by juzhiyuan** [reply]
> 我是第一次搞这些，也不太清楚流程和规范，如果方便的话，可以加个微信交流一下吗

Hi, my wechat is JuShaoyao, but I still prefer to suggest using the official mailing list and GitHub issue to discuss and track issues.
